src/MonoMod.Utils/Cil/ILCursor.cs

```diff
@@ -460,40 +478,63 @@ public ILCursor GotoLabel(ILLabel label, MoveType moveType = MoveType.AfterLabel
         public ILLabel DefineLabel() => Context.DefineLabel();
 
         private ILCursor _Insert(Instruction instr) {
+            // retargetting
+            if (_afterLabels != null)
+                foreach (var label in _afterLabels)
+                    label.Target = instr;
+
+            if (_afterHandlerStarts) {
+                foreach (var eh in Body.ExceptionHandlers) {
+                    if (eh.TryStart == Next) eh.TryStart = instr;
+                    if (eh.HandlerStart == Next) eh.HandlerStart = instr;
+                    if (eh.FilterStart == Next) eh.FilterStart = instr;
+                }
+            }
+
+            if (_afterHandlerEnds) {
+                foreach (var eh in Body.ExceptionHandlers) {
+                    if (eh.TryEnd == Next) eh.TryEnd = instr;
+                    if (eh.HandlerEnd == Next) eh.HandlerEnd = instr;
+                }
+            }
```

Comment:
EH end markers point to the instruction following the end of the range, so they are not inclusive. When inserting an instruction after an EH range, the markers need to be updated to point to it, otherwise the inserted instruction would be inside the EH range.

Eg
```cs
try { ... } 
catch { ... 
    Prev
    // cursor is not here (before EH end)
}
// cursor is here (after EH end)
Next
```

Reply:
Ah, OK.